### ISSUE TITLE
nvim_tree_lsp_diagnostics switch reversed

### ISSUE THREAD

**Issue by wayjam** [kyazdani42/nvim-tree.lua#634]
set `nvim_tree_lsp_diagnostics = 0` not works, icons still shown

![image](https://user-images.githubusercontent.com/2463831/134777480-ea7fa0b5-26a2-4cb3-b9ef-784f9467b5f3.png)

**Comment by xiyaowong** [comment]
you should provide more information about this bug

**Comment by xiyaowong** [comment]
https://github.com/kyazdani42/nvim-tree.lua/blob/master/lua/nvim-tree.lua#L43

currently, some options are not supported to be set using setup

**Comment by wayjam** [comment]
refer to https://github.com/kyazdani42/nvim-tree.lua/blob/master/doc/nvim-tree-lua.txt#L320

```
|g:nvim_tree_lsp_diagnostics|                  *g:nvim_tree_lsp_diagnostics*

Can be 0 or 1. When 1, will show nvim-lsp diagnostics in the signcolumn
of the tree highlighted by diagnostic severity.
Code will be executed on `LspDiagnosticsChanged`. 0 by default.
```

But not `  if vim.g.nvim_tree_lsp_diagnostics ~= 1 then` would trigger update.

**Comment by xiyaowong** [comment]
sorry hhh😂

**Comment by siduck76** [comment]
im facing this issue too :c

**Comment by kyazdani42** [comment]
should be fixed in `1.3.1` (d4172928e2725666)
